### ISSUE TITLE
Improve Screen Reader Support

### ISSUE THREAD

**Issue by slannon97** [BuddiesOfBudgie/budgie-desktop#737]
I'm totally blind and use the Orca screen reader. I recently tried Budgie 10.9.2 on Arch Linux, and in some ways, I prefer it to MATE, which is my current desktop in Linux due to its accessibility and ease of use. However, there are accessibility problems that should be addressed. Here are just a few.

Orca doesn't correctly speak the names of application windows if you hold down the Alt key and press Tab. Instead, it constantly speaks the name of the application you're in, rather than correctly reading which window you'll move to. The only way to know what you've switched to is if you let go of Alt.
Pressing Super and d is supposed to take you to the desktop, but this either doesn't work properly, or Orca doesn't read anything. The desktop icons should be usable with the screen reader.
The Budgie Control Center doesn't work with Orca. Tabbing around and/or using the arrow keys either speaks nothing at all, or something about a search and primary menu toggle button. However, it's not clear what these do. Other applications such as the Nemo file manager and Budgie Desktop Settings don't appear to work well either. IN general, it should be possible to tab and/or use the arrow keys in these applications and receive clear feedback about controls you're moving to.

The widgets and notification panels don't read very well or at all with Orca. It says button or nothing at all, which indicates unlabeled controls and controls that Orca can't access at all. Again, navigating using the keyboard should allow the user to select controls and clearly understand what's going on. This is especially important, since as far as I know, the widgets panel is like the Windows System tray where you access important information and settings for things like Wi-Fi, Bluetooth, volume, etc. Currently, screen reader users are locked out.

The application launcher is almost perfect. however, here are a few suggestions to make it better for a keyboard and/or screen reader user.
The menu should be a little more predictable when using a keyboard. For example, allow typing the first letter to jump to an option when focused in a menu. The categories should be arranged vertically and be navigable with up/down arrow. Pressing right arrow or enter should go into a category and display items in that category that can be navigated using up/down arrow keys. Pressing left arrow should return to the list of categories. Pressing tab should move to another menu with the options for your user account and the power button. Currently, the button labeled with my username appears to do nothing when pressing space bar or enter on it. The power menu should be a vertical list you can navigate with up/down arrow.

Essentially, the way I imagine this working is something like this. Press the Windows/Super key to bring up the search box. This works well right now. Either type what you're looking for and use up/down to browse the results, or ignore the search box and press down arrow to get into the menus. Tab should switch between the list for the menus, the search box, and the list with power options, etc.

If these changes can be made, this would be a great desktop to use as a blind user. If you have any questions, please don't hesitate to ask, and I'll do my best to provide as much information as I can. I'm not a programmer, but I want this desktop to improve, so hopefully someone will prioritize accessibility.

**Comment by joanmarie** [comment]
Orca maintainer here. 👋 

There are indeed some accessibility issues. Some greater than others.

I just landed a change in Orca's main and gnome-49 branches to support the Alt+Tab switcher, which is accessible enough. It works with (at least) v. 10.9.3 which is what is currently in Fedora 43.

The input that pops up when you press F2, along with its associated popup, appear to be completely inaccessible:
* There is no window claiming activation
* There are no focus change events
* There are no selection change events

In other words, when one presses F2, Orca only knows that the previously-active window is now inactive. It has no clue that anything else is taking place and thus is silent.

**Comment by joanmarie** [comment]
Regarding notification presentation: It would be great if you could fire an `object:state-changed:showing` event and give the notification popup a role of `notification`. Orca should then do some work to figure out the contents.
